backend/app/organisation/orders_handlers.go

```diff
@@ -0,0 +1,338 @@
+			return
+		}
+
+		articlesID := make([]uint, len(orderArticles))
+		for orderArticleIndex := 0; orderArticleIndex < len(orderArticles); orderArticleIndex++ {
+			articlesID = append(articlesID, orderArticles[orderArticleIndex].ArticleID)
+		}
+
+		articles, err := GetArticles(articlesID)
+		if err != nil {
+			return
+		}
+
+		for orderArticleIndex := 0; orderArticleIndex < len(orderArticles); orderArticleIndex++ {
+			// search for articles
+			article := Articles{}
+			for articleIndex := 0; articleIndex < len(articles); articleIndex++ {
+				if articles[articleIndex].ID == orderArticles[orderArticleIndex].ArticleID {
+					article = articles[articleIndex]
+					break
+				}
+			}
+
+			pictures, err := GetArticlePictures(article.ID)
+			if err != nil {
+				return
+			}
+			ordersPresenter[i].Items = append(ordersPresenter[i].Items, &OrderItemPresenter{
+				OrdersArticles: &orderArticles[orderArticleIndex],
+				Article: ArticlesPresenter{
+					Articles: article,
+					Pictures: pictures,
+				},
+			})
+		}
+	}
+
+	ctx.JSON(http.StatusOK, map[string]interface{}{
+		"limit":  limit,
+		"offset": offset,
+		"data":   ordersPresenter,
+	})
+}
+
+func GetUserOrdersHandler(ctx *gin.Context) {
+
+	// TODO Retreive user information an user the same flow in GetOrganisationOrdersHandler
+}
+
+func GetOrderHandler(ctx *gin.Context) {
+
+	/*	organisationID, err := strconv.Atoi(ctx.GetHeader("Tenant"))
+		if err != nil {
+			ctx.String(http.StatusBadRequest, ErrTenantNotProvided.Error())
+			return
+		}*/
+
+	orderID, err := strconv.Atoi(ctx.Param("orderID"))
+	if err != nil {
+		ctx.String(http.StatusBadRequest, "Order id must be an int")
+		return
+	}
+
+	// TODO check if the issuer of this request is the owner of the order or a manager of the organisation
+
+	order, err := GetOrder(uint(orderID))
+	if err != nil {
+		return
+	}
+	orderPresenter := OrderPresenter{}
+	orderPresenter.Orders = order
+
+	orderArticles, err := GetOrderArticles(order.ID)
+	if err != nil {
+		return
+	}
+
+	articlesID := make([]uint, len(orderArticles))
+	for orderArticleIndex := 0; orderArticleIndex < len(orderArticles); orderArticleIndex++ {
+		articlesID = append(articlesID, orderArticles[orderArticleIndex].ArticleID)
+	}
+
+	articles, err := GetArticles(articlesID)
+	if err != nil {
+		return
+	}
+
+	for orderArticleIndex := 0; orderArticleIndex < len(orderArticles); orderArticleIndex++ {
+		// search for articles
+		article := Articles{}
+		for articleIndex := 0; articleIndex < len(articles); articleIndex++ {
+			if articles[articleIndex].ID == orderArticles[orderArticleIndex].ArticleID {
+				article = articles[articleIndex]
+				break
+			}
+		}
+
+		pictures, err := GetArticlePictures(article.ID)
+		if err != nil {
+			return
+		}
+		orderPresenter.Items = append(orderPresenter.Items, &OrderItemPresenter{
+			OrdersArticles: &orderArticles[orderArticleIndex],
+			Article: ArticlesPresenter{
+				Articles: article,
+				Pictures: pictures,
+			},
+		})
+	}
```

Comment:
Let's try to use the `for _, elem := range elems { ` as much as possible

Reply:
resolved